wordpress-mobile/WordPress-iOS
# Domains: Fix app hanging when selecting domain by showing an error notice

Fixes #20748

To test:

After choosing a domain e.g. hometitle.org, the app should advance to the checkout screen to allow me to complete the purchase.

1. Tap the Find a custom domain dashboard card
2. Tap Search for a domain
3. Tap the first domain suggestion
4. Tap Select domain
5. Confirm that error notice appears and spinner disappears

## Regression Notes
1. Potential unintended areas of impact

None, we're adding additional handling to before unhandled code path

2. What I did to test those areas of impact (or what existing automated tests I relied on)

None

3. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:


UI Changes testing checklist:

https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/bb271df9-85de-495f-a497-36e827b0dab6

![image](https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/314cdce8-fe32-4e80-85b0-fdbc185c9ead)
